src/main/java/com/exactpro/th2/sim/impl/SimulatorRuleInfo.java

```diff
@@ -79,7 +79,17 @@ public void send(@NotNull Message msg) {
         if (LOGGER.isTraceEnabled()) {
             LOGGER.trace("Process message by rule with ID '{}' = {}", id, TextFormat.shortDebugString(msg));
         }
-        String sessionAlias = StringUtils.defaultIfEmpty(msg.getMetadata().getId().getConnectionId().getSessionAlias(), this.sessionAlias);
+
+        String sessionAlias = msg.getMetadata().getId().getConnectionId().getSessionAlias();
+
+        if (StringUtils.isEmpty(sessionAlias)) {
+            sessionAlias = this.sessionAlias;
+
+            Message.Builder builder = msg.toBuilder();
+            builder.getMetadataBuilder().getIdBuilder().getConnectionIdBuilder().setSessionAlias(this.sessionAlias);
```

Comment:
Should be the local variable I guess. It doesn't change anything in the current case but if you modify the session alias somehow you might forget to set it here
```suggestion
            builder.getMetadataBuilder().getIdBuilder().getConnectionIdBuilder().setSessionAlias(sessionAlias);
```